frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -371,84 +379,162 @@ function BookInfo:setCustomBookCover(file, book_props, metadata_updated_caller_c
-                        words_nb = words_nb - 1
-                    end
-                end
-            end
+        custom_doc_settings:saveSetting("custom_props", custom_props)
+        custom_doc_settings:flushCustomMetadata(file)
+    end
+    book_props.display_title = book_props.display_title or display_title -- restore
+    -- in memory
+    prop_value = prop_value or custom_doc_settings:readSetting("doc_props")[prop_key] -- set custom or restore original
+    book_props[prop_key] = prop_value
+    if self.ui then -- currently opened document
+        self.ui.doc_props[prop_key] = prop_value
+        if prop_key == "title" then -- generate if original is empty
+            self.ui.doc_props.display_title = prop_value or filemanagerutil.splitFileNameType(file)
         end
     end
-    return lines_nb, words_nb
+    self:updateBookInfo(file, book_props, metadata_updated_caller_callback, prop_key)
+end
+
+function BookInfo:showCustomEditDialog(file, book_props, metadata_updated_caller_callback, prop_key)
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = _("Edit book property: ") .. self.prop_text[prop_key]:sub(1, -2),
+        input = book_props[prop_key],
+        input_type = prop_key == "series_index" and "number",
+        allow_newline = prop_key == "authors" or prop_key == "keywords" or prop_key == "description",
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(input_dialog)
+                    end,
+                },
+                {
+                    text = _("Save"),
+                    callback = function()
+                        local prop_value = input_dialog:getInputValue()
+                        if prop_value and prop_value ~= "" then
+                            UIManager:close(input_dialog)
+                            self:setCustomMetadata(file, book_props, metadata_updated_caller_callback, prop_key, prop_value)
+                        end
+                    end,
+                },
+            },
+        },
+    }
+    UIManager:show(input_dialog)
+    input_dialog:onShowKeyboard()
 end
 
-function BookInfo:showCustomMenu(file, book_props, metadata_updated_caller_callback)
+function BookInfo:showCustomDialog(file, book_props, metadata_updated_caller_callback, prop_key)
+    local original_prop, custom_prop, prop_is_cover
+    if prop_key then -- metadata
+        if self.custom_doc_settings then
+            original_prop = self.custom_doc_settings:readSetting("doc_props")[prop_key]
+            custom_prop = self.custom_doc_settings:readSetting("custom_props")[prop_key]
+        else
+            original_prop = book_props[prop_key]
+        end
+        if original_prop and prop_key == "description" then
+            original_prop = util.htmlToPlainTextIfHtml(original_prop)
```

Comment:
No, currently `BookInfo:showCustomEditDialog()` takes unstripped html description.
Should be fixed.

I see that we do not use original HTML'ed description anywhere.
Maybe clean it at the source, in `Document:getProps()`?

Reply:
> I see that we do not use original HTML'ed description anywhere.
> Maybe clean it at the source, in `Document:getProps()`?

My preference in general is to save the orignal stuff as it is, original :)
If we later fix/tweak the cleaning code, it will apply immediately, without the need to refresh book metadata.
And maybe one day we'll want to display it as HTML (I think there was an issue about it - the resulting text being bad, that we solved another way).

But then I don't know how this plays with "custom description" :/ I guess the custom description should be plain text - dunno if where the htmlToPlainText stuff happens is too late to know about that.